docs/operator/resourcesets/github-pull-requests.md

```diff
@@ -249,6 +250,78 @@ spec:
     region: "us-east-1"
 ```
 
+### Status reporting on Pull Requests
+
+To notify the developers of the preview deployment status, we can use the
+Flux [GitHub Dispatch Provider](https://fluxcd.io/flux/components/notification/providers/#github-dispatch)
+to post a comment on the PR page with the preview URL and the latest Helm release status.
+
+First we create a Flux Provider and Alert in the `app-preview` namespace:
+
+```yaml
+---
+apiVersion: notification.toolkit.fluxcd.io/v1beta3
+kind: Provider
+metadata:
+  name: github-dispatch
+  namespace: app-preview
+spec:
+  type: githubdispatch
+  address: https://github.com/org/app
+  secretRef:
+    name: github-auth
+---
+apiVersion: notification.toolkit.fluxcd.io/v1beta3
+kind: Alert
+metadata:
+  name: github-pr-comment
+  namespace: app-preview
+spec:
+  providerRef:
+    name: github-dispatch
+  eventSources:
+    - kind: HelmRelease
+      name: '*'
+```
+
+Then in the GitHub repository, we need to define a GitHub Workflow that parses the Flux event
+metadata and posts a comment on the PR page:
+
+```yaml
+name: flux-preview-comment
+on:
+  repository_dispatch:
+jobs:
+  comment:
+    if: github.event.client_payload.metadata.pr-number != ''
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+    steps:
+      - name: Compose Comment
+        run: |
+          tee comment.md <<'EOF'
+          Flux deployment ${{ github.event.client_payload.severity }}:
+          - Preview URL: ${{ github.event.client_payload.metadata.preview-url }}
+          - Revision: ${{ github.event.client_payload.metadata.revision }}
+          - Status: ${{ github.event.client_payload.message }}
+          EOF
+      - name: Post Comment
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          GITHUB_REPO: ${{ github.repository }}
+          PR_NUMBER: ${{ github.event.client_payload.metadata.pr-number }}
+        run: |
+          gh pr comment $PR_NUMBER \
+          --repo $GITHUB_REPO \
+          --body-file comment.md \
+          --create-if-none \
+          --edit-last
```

Comment:
Note, this flag is defined like this:

```go
cmd.Flags().BoolVar(&opts.EditLast, "edit-last", false, "Edit the last comment of the same author")
```

So this is a bit limited, comments are keyed by author. Multiple Flux objects e.g. multiple Kustomizations and/or HelmReleases would race for updating this comment.

This fact adds up to the reasons for implementing :+1: https://github.com/fluxcd/notification-controller/issues/1073